tests/kernel/x86boot_page_table/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+-   test:
+        platform_whitelist: qemu_x86
```

Comment:
Is there any reason this test will not run on any x86 platform?

Seems like we should have:

arch_whitelist: x86

for sure in here.